src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/DynamicSSHKeyInjectionInstanceType.tsx

```diff
@@ -2,17 +2,20 @@ import React from 'react';
 
 import { useInstanceTypeVMStore } from '@catalog/CreateFromInstanceTypes/state/useInstanceTypeVMStore';
 import { instanceTypeActionType } from '@catalog/CreateFromInstanceTypes/state/utils/types';
-import { DYNAMIC_CREDENTIALS_SUPPORT } from '@kubevirt-utils/components/DynamicSSHKeyInjection/constants/constants';
 import { DynamicSSHKeyInjection } from '@kubevirt-utils/components/DynamicSSHKeyInjection/DynamicSSHKeyInjection';
 import DynamicSSHKeyInjectionHelpTextIcon from '@kubevirt-utils/components/DynamicSSHKeyInjection/DynamicSSHKeyInjectionHelpTextIcon';
 import VirtualMachineDescriptionItem from '@kubevirt-utils/components/VirtualMachineDescriptionItem/VirtualMachineDescriptionItem';
+import useDataSourceForVolumeSnapshot from '@kubevirt-utils/hooks/useDataSourceForVolumeSnapshot';
 import { t } from '@kubevirt-utils/hooks/useKubevirtTranslation';
 
+import { dataSourceHasDynamicSSHLabel } from '../utils/utils';
+
 const DynamicSSHKeyInjectionInstanceType = () => {
   const { instanceTypeVMState, setInstanceTypeVMState } = useInstanceTypeVMStore();
-  const { sshSecretCredentials } = instanceTypeVMState;
-  const hasDynamicSSHLabel =
-    instanceTypeVMState?.pvcSource?.metadata?.labels?.[DYNAMIC_CREDENTIALS_SUPPORT];
+  const { pvcSource, sshSecretCredentials, volumeSnapshotSource } = instanceTypeVMState;
+  const dataSource = useDataSourceForVolumeSnapshot(volumeSnapshotSource);
```

Comment:
instanceTypesVMState has a selectedVolume isnt it the dataSource in this case?

Reply:
That worked. =) I didn't know I could get the bootable volume there. Thanks!